presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -185,9 +187,11 @@ public PlanNode replaceChildren(List<PlanNode> newChildren)
 
     public boolean isDecomposable(FunctionRegistry functionRegistry)
     {
-        return getAggregations().entrySet().stream()
+        return (getAggregations().entrySet().stream()
                 .map(entry -> functionRegistry.getAggregateFunctionImplementation(entry.getValue().getSignature()))
-                .allMatch(InternalAggregationFunction::isDecomposable);
+                .allMatch(InternalAggregationFunction::isDecomposable)) &&
+                getAggregations().entrySet().stream()
+                        .allMatch(entry -> entry.getValue().getOrderBy().isEmpty() && entry.getValue().getOrdering().isEmpty());
```

Comment:
Is it just enough to test `getOrderBy().isEmpty()`?
Maybe add utility method `hasOrderBy` to `Aggregation`. You have this check in various places (checkStates).